It makes no sense to do PCIe WA window setup in the individual
board support files while the decision whether or not to use the
PCIe WA access method is made in a different place, in the PCIe
support code.

This patch moves the configuration of a PCIe WA window from the
individual Orion board support files to the central Orion PCIe
support code.

Signed-off-by: Lennert Buytenhek <buytenh@marvell.com>
---
 arch/arm/mach-orion5x/dns323-setup.c      |    7 -------
 arch/arm/mach-orion5x/kurobox_pro-setup.c |    6 ------
 arch/arm/mach-orion5x/pci.c               |    2 ++
 arch/arm/mach-orion5x/rd88f5182-setup.c   |    6 ------
 arch/arm/mach-orion5x/ts209-setup.c       |    6 ------
 5 files changed, 2 insertions(+), 25 deletions(-)

--- a/arch/arm/mach-orion5x/dns323-setup.c
+++ b/arch/arm/mach-orion5x/dns323-setup.c
@@ -247,13 +247,6 @@
 	/* Setup basic Orion functions. Need to be called early. */
 	orion5x_init();
 
-	/* DNS-323 has a Marvell 88X7042 SATA controller attached via PCIe
-	 *
-	 * Open a special address decode windows for the PCIe WA.
-	 */
-	orion5x_setup_pcie_wa_win(ORION5X_PCIE_WA_PHYS_BASE,
-				ORION5X_PCIE_WA_SIZE);
-
 	/* set MPP to 0 as D-Link's 2.6.12.6 kernel did */
 	orion5x_write(MPP_0_7_CTRL, 0);
 	orion5x_write(MPP_8_15_CTRL, 0);
--- a/arch/arm/mach-orion5x/kurobox_pro-setup.c
+++ b/arch/arm/mach-orion5x/kurobox_pro-setup.c
@@ -188,12 +188,6 @@
 	orion5x_init();
 
 	/*
-	 * Open a special address decode windows for the PCIe WA.
-	 */
-	orion5x_setup_pcie_wa_win(ORION5X_PCIE_WA_PHYS_BASE,
-				ORION5X_PCIE_WA_SIZE);
-
-	/*
 	 * Setup Multiplexing Pins --
 	 * MPP[0-1] Not used
 	 * MPP[2] GPIO Micon
--- a/arch/arm/mach-orion5x/pci.c
+++ b/arch/arm/mach-orion5x/pci.c
@@ -152,6 +152,8 @@
 	if (dev == MV88F5181_DEV_ID || dev == MV88F5182_DEV_ID) {
 		printk(KERN_NOTICE "Applying Orion-1/Orion-NAS PCIe config "
 				   "read transaction workaround\n");
+		orion5x_setup_pcie_wa_win(ORION5X_PCIE_WA_PHYS_BASE,
+					  ORION5X_PCIE_WA_SIZE);
 		pcie_ops.read = pcie_rd_conf_wa;
 	}
 
--- a/arch/arm/mach-orion5x/rd88f5182-setup.c
+++ b/arch/arm/mach-orion5x/rd88f5182-setup.c
@@ -249,12 +249,6 @@
 	orion5x_init();
 
 	/*
-	 * Open a special address decode windows for the PCIe WA.
-	 */
-	orion5x_setup_pcie_wa_win(ORION5X_PCIE_WA_PHYS_BASE,
-				ORION5X_PCIE_WA_SIZE);
-
-	/*
 	 * Setup Multiplexing Pins --
 	 * MPP[0] Debug Led (GPIO - Out)
 	 * MPP[1] Debug Led (GPIO - Out)
--- a/arch/arm/mach-orion5x/ts209-setup.c
+++ b/arch/arm/mach-orion5x/ts209-setup.c
@@ -365,12 +365,6 @@
 	orion5x_init();
 
 	/*
-	 * Open a special address decode windows for the PCIe WA.
-	 */
-	orion5x_setup_pcie_wa_win(ORION5X_PCIE_WA_PHYS_BASE,
-				ORION5X_PCIE_WA_SIZE);
-
-	/*
 	 * Setup Multiplexing Pins --
 	 * MPP[0] Reserved
 	 * MPP[1] USB copy button (0 active)
